### ISSUE TITLE
Issues with codegen generating PUBLIC always

### ISSUE THREAD

**Issue by Zidane** [jOOQ/jOOQ#7650]
### Expected behavior and actual behavior:
I am using codegen to build a model off of some scripts (using DDLDatabase). My database name is unknown, entirely, until runtime (configurable by the user). I found some documentation that states I need to set `outputSchemaToDefault = true` but my generated code still contains a generated PUBLIC schema and accessing any values from a `Result` object using the `Table` field objects still generate queries that start with PUBLIC.

I'd appreciate any assistance or point out what I'm doing wrong...

Gradle script:

```
    titleDdl(sourceSets.main) {
        generator {
            database {
                name = 'org.jooq.meta.extensions.ddl.DDLDatabase'
                properties {
                    property {
                        key = 'scripts'
                        value = 'src/main/resources/db/title/database.sql'
                    }
                }
                inputCatalog = ''
                outputCatalogToDefault = true
                outputSchema = ''
                outputSchemaToDefault = true
            }
            generate {
                relations = true
                deprecated = false
                records = true
                immutablePojos = true
                fluentSetters = true
            }
            target {
                packageName = 'com.almuradev.generated.title'
                directory = generatedDir
            }
        }
    }
```

### Versions:

- jOOQ: 3.11.0
- Java: 1.8
- Database (include vendor): h2
- OS: Windows 8.1
- JDBC Driver (include name if inofficial driver): h2

**Comment by Zidane** [comment]
If I manually go to the table in question and change:

```java
    /**
     * {@inheritDoc}
     */
    @Override
    public Schema getSchema() {
        return Public.PUBLIC;
    }
```

To return ```null``` instead this drops the PUBLIC moniker but obviously do not want to be touching generated classes >.<

**Comment by lukaseder** [comment]
Thank you very much for your report. As a workaround, have you tried adding an explicit `inputSchema`? E.g. `inputSchema = 'PUBLIC'`?

This definitely should be improved. I'm assuming that all your tables in `database.sql` are unqualified?

**Comment by lukaseder** [comment]
> Is it a problem that constraints still appear to do PUBLIC.X.X

What do you mean by this?

**Comment by Zidane** [comment]
@lukaseder 

INFO: Adding foreign key       : CONSTRAINT_33 (PUBLIC.title_select.title) referencing CONSTRAINT_6

**Comment by lukaseder** [comment]
That's the `Definition.inputName` not the `Definition.outputName`, so that's consistent with the current behaviour.

**Comment by Zidane** [comment]
@lukaseder 

Sounds good then, thanks for the insight!

**Comment by lukaseder** [comment]
This is fixed now for jOOQ 3.12 through #7759. The workaround in 3.11 is to use the schema mapping feature.
